### ISSUE TITLE
Render dropdowns in their own top-level layer

### ISSUE THREAD

**Issue by vladimir-rovensky** [jquense/react-widgets#132]
Problem: When you place the DropDown component within a container with overflow:auto and constrained width and/or height, the actual dropdown part is rendered within the container, rather than on top of it. If it doesn't quite fit within the container, it is not visible (you can't select the items) and the scrollbar(s) appear on the container, which looks very ugly.

Other frameworks (e.g. Kendo UI) solve this issue by rendering the dropdowns in their own container at the document root, would it be possible to do the same in this library?

**Comment by vladimir-rovensky** [comment]
I tried to attach a screenshot with the problem, but GitHub won't let me for some reason.
It's rather simple to reproduce the issue on the examples page (http://jquense.github.io/react-widgets/docs/#/dropdown-list), just give the .form-group overflow:auto and height: 100px.

**Comment by jquense** [comment]
Hi there thanks for the issue.

I've actually intentionally chosen to render popups as children of their respective widgets, over going the kendoUI approach of using `position: fixed` popups appended to the body. In general I've found the current approach ends up with fewer issues for extending and working with components at a lower level. It also involves a lot less complexity, popups can be positioned solely through css, and not need for extensive global listeners for window scrolling, and hard to calculate positions.

Which is all fine and good but doesn't actually address your problem. I mention it because this issue is the big "con" to this approach and one I am aware of, but think worth it given the other benefits. In terms of fixing it, setting `overflow: visible` or `scroll` on your containers will fix the issue in different ways. In most cases I found this is fine, but I agree that in some cases you do actually want `overflow: auto` and it will be sort of ugly.
